.gitignore

```diff
@@ -15,12 +15,44 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
-# OSX
+# macOS
 .DS_Store
 
 # Bazel
 bazel-*
 
-# IntelliJ
-.idea
-.ijwb
+# IntelliJ IDEA
+.idea/
+*.iml
+*.ipr
+*.iws
+*.geany
+.ijwb/
+
+# TypeDB Studio
+.typedb-studio
+
+# VS Code
+.vscode/
+.settings/
+.project
+.classpath
+
+# Eclipse
+.metadata/
+
+# Compiled
+dist/
+out/
+*.class
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml #
+hs_err_pid*
+
+# Log
+logs/
+*.log
```

Comment:
I know I've seen `.log` files generated in projects before, but can't quite remember the environment / context... any ideas?

Reply:
I think `.log` is pretty generic, but we can scrap it if we're sure no IDE or OS we seek to cover is creating them.

Reply:
I have the same gut feeling. Let's keep it and move forward.